File: app/src/main/java/com/kakacat/minitool/inquireip/InquireIpActivity.java
package com.kakacat.minitool.inquireip;

import android.content.Context;
import android.os.Bundle;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.textfield.TextInputLayout;
import com.kakacat.minitool.R;
import com.kakacat.minitool.common.util.UiUtil;

public class InquireIpActivity extends AppCompatActivity implements Contract.View {

    private Contract.Presenter presenter;

    private TextView tvCountry;
    private TextView tvProvince;
    private TextView tvCity;
    private TextView tvIsp;
    private TextView tvIp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_inquire_ip);

        initData();
        initView();
    }


    @Override
    public void initData() {
        getPresenter();
    }

    @Override
    public void initView() {
        initToolbar();

        TextInputLayout til = findViewById(R.id.text_input_layout);
        EditText editText = findViewById(R.id.et_input_ip);
        tvCountry = findViewById(R.id.tv_country);
        tvProvince = findViewById(R.id.tv_province);
        tvCity = findViewById(R.id.tv_city);
        tvIsp = findViewById(R.id.tv_isp);
        tvIp = findViewById(R.id.tv_ip);

        til.setStartIconOnClickListener(view -> presenter.requestIpData(editText.getText().toString()));
    }

    @Override
    public Contract.Presenter getPresenter() {
        if (presenter == null) {
            presenter = new Presenter(this);
        }
        return presenter;
    }

    @Override
    public void onUpdateDataCallBack(IpBean ipBean, String result) {
        UiUtil.showToast(this, result);
        if (ipBean != null) {
            tvCountry.setText(ipBean.getCountry());
            tvProvince.setText(ipBean.getProvince());
            tvCity.setText(ipBean.getCity());
            tvIsp.setText(ipBean.getIsp());
            tvIp.setText(ipBean.getIpAddress());
        }
    }

    private void initToolbar() {
        setSupportActionBar(findViewById(R.id.toolbar_inquire_ip));
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeAsUpIndicator(R.drawable.ic_action_back);
            actionBar.setDisplayShowTitleEnabled(false);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem menuItem) {
        if (menuItem.getItemId() == android.R.id.home) {
            finish();
        }
        return true;
    }

    @Override
    public Context getContext() {
        return this;
    }


}

File: app/src/main/java/com/kakacat/minitool/inquireip/InquireIpActivity.kt
package com.kakacat.minitool.inquireip

import android.os.Bundle
import android.widget.EditText
import android.widget.TextView
import com.google.android.material.textfield.TextInputLayout
import com.kakacat.minitool.R
import com.kakacat.minitool.common.base.BaseActivity
import com.kakacat.minitool.common.util.UiUtil.initToolbar
import com.kakacat.minitool.common.util.UiUtil.showToast

class InquireIpActivity : BaseActivity(), Contract.View {

    private var presenter: Contract.Presenter? = null

    private val tvCountry by lazy { findViewById<TextView>(R.id.tv_country) }
    private val tvProvince by lazy { findViewById<TextView>(R.id.tv_province) }
    private val tvCity by lazy { findViewById<TextView>(R.id.tv_city) }
    private val tvIsp by lazy { findViewById<TextView>(R.id.tv_isp) }
    private val tvIp by lazy { findViewById<TextView>(R.id.tv_ip) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_inquire_ip)
        initData()
        initView()
    }

    override fun onDestroy() {
        presenter = null
        super.onDestroy()
    }

    override fun initData() {
        presenter = Presenter(this)
    }

    override fun initView() {
        initToolbar(this, false)
        val til = findViewById<TextInputLayout>(R.id.text_input_layout)
        val editText = findViewById<EditText>(R.id.et_input_ip)
        til.setStartIconOnClickListener { presenter!!.requestIpData(editText.text.toString()) }
    }

    override fun onUpdateDataCallBack(ipBean: IpBean?, result: String?) {
        showToast(this, result)
        if (ipBean != null) {
            tvCountry.text = ipBean.country
            tvProvince.text = ipBean.province
            tvCity.text = ipBean.city
            tvIsp.text = ipBean.isp
            tvIp.text = ipBean.ipAddress
        }
    }
}